### PR TITLE
TableTop microapp improvements

### PR DESCRIPTION
**Pull request by gunt0001**
Related to issue: https://devtopia.esri.com/runtime/kotlin/issues/5013

### Description:

Changes the microapp to identify buildings in the scene and display their IDs in the callout

### Summary of changes:

- Add `IdentifiedBuilding` data class and `ArcGISSceneLayer.identifyBuilding` extension function to get a feature and location in the scene and bundle them together
- Change the callout in the `TableTopSceneView` content lambda to display the building info
- Move some code around and add comments for readability

### Pre-merge Checklist

- a [vTest](https://runtime-kotlin.esri.com/view/all/job/vtest/job/toolkit/) Job for this PR has been run
  - [ ] link: https://runtime-kotlin.esri.com/view/all/job/vtest/job/toolkit/203/
- Unit and/or integration tests have been added to exercise this PR's logic, and the tests are passing:
  - [ ] Yes
  - [x] No

### CODE REVIEW

**Review by gunt0001 — COMMENTED** [review]
@hud10837 - looks great. I did some sanity testing against SDK build 4424, which worked well.

File: microapps/ArTabletopApp/app/src/main/java/com/arcgismaps/toolkit/artabletopapp/screens/MainScreen.kt [hunk 1]
```diff
@@ -82,61 +92,69 @@ fun MainScreen() {
                 initializationStatus = it
             },
             onSingleTapConfirmed = {
-                val location = tableTopSceneViewProxy.screenToBaseSurface(it.screenCoordinate)
-                location?.let { point ->
-                    tappedLocation = point
+                arcGISSceneLayer.clearSelection()
+                coroutineScope.launch {
+                    identifiedBuilding = arcGISSceneLayer.identifyBuilding(
+                        it.screenCoordinate,
```
Comment (01smito01) [inline microapps/ArTabletopApp/app/src/main/java/com/arcgismaps/toolkit/artabletopapp/screens/MainScreen.kt:98]: It's not immediately obvious what "it" is in this context - would this be better as a named parameter (eg "tap")?

File: microapps/ArTabletopApp/app/src/main/java/com/arcgismaps/toolkit/artabletopapp/screens/MainScreen.kt [hunk 2]
```diff
@@ -158,3 +176,31 @@ fun TextWithScrim(text: String) {
...
+ * this function returns `null`.
+ *
+ * @since 200.6.0
+ */
+private suspend fun ArcGISSceneLayer.identifyBuilding(
+    screenCoordinate: ScreenCoordinate,
+    proxy: TableTopSceneViewProxy
+): IdentifiedBuilding? {
+    val identifyLayerResult = proxy.identify(this, screenCoordinate, 50.dp).getOrElse {
+        Log.i("ArTabletopApp", "Failed to identify building: $it")
```
Comment (01smito01) [inline microapps/ArTabletopApp/app/src/main/java/com/arcgismaps/toolkit/artabletopapp/screens/MainScreen.kt:192]: I've had a play around with the app and I've not seen this message in logcat, so it seems pretty unusual. Should this be logged as a warning?

Comment (hud10837) [reply microapps/ArTabletopApp/app/src/main/java/com/arcgismaps/toolkit/artabletopapp/screens/MainScreen.kt]: I also couldn't get it to happen, so I've just removed the log

**Comment by hud10837** [comment]
Thanks @01smito01 for the quick review. Merging now

**Pull request merged**